File: src/main/java/kr/co/lunatalk/domain/productlike/repository/ProductLikeRepository.java
package kr.co.lunatalk.domain.productlike.repository;

import kr.co.lunatalk.domain.productlike.domain.ProductLike;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.Optional;

public interface ProductLikeRepository extends JpaRepository<ProductLike, Long>, ProductLikeRepositoryCustom {

	Optional<ProductLike> findByMemberIdAndProductId(Long memberId, Long productId);

	void deleteByMemberIdAndProductId(Long memberId, Long productId);
}

File: lunatalk-core/src/main/kotlin/kr/co/lunatalk/domain/productlike/repository/ProductLikeRepository.kt
package kr.co.lunatalk.domain.productlike.repository

import kr.co.lunatalk.domain.productlike.domain.ProductLike
import org.springframework.data.jpa.repository.JpaRepository
import java.util.Optional

interface ProductLikeRepository : JpaRepository<ProductLike, Long>, ProductLikeRepositoryCustom {

    fun findByMemberIdAndProductId(memberId: Long, productId: Long): Optional<ProductLike>

    fun deleteByMemberIdAndProductId(memberId: Long, productId: Long)
}
